the-collab-lab/tcl-55-smart-shopping-list
# Adds Chakra UI To List and ListItem Components

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description
This code changes the styling structure of all of the `List` and `ListItem` components by using chakra UI components.  It also adds a new component, `Dialog`, to replace the window.confirm dialog. This code also adds a new display for the user's current token and the ability to quickly copy the token.

Link to the design wireframe: https://www.figma.com/file/4oxU5wehZPNinZpr24UuLw/Chakra-UI-Figma-Kit-(Community)?type=design&node-id=2102-2110&t=ciT6mhpJqrDlx0bY-0 

## Related Issue

Closes #34 
Closes #35 

## Acceptance Criteria

### Issue 16.
- [x] Header should be fixed and contain the List Luxe title.
- [x] When the user joins a new list or creates one and navigate directly to `/list`, the current token should be displayed along with a graphic to copy the token on click (Asset: `CkCopy`chakra icon)
- [x] The `Search` input field should include an `x` to clear the search filter. (Asset: `InputRightElement` with `CkClose` chakra icon)
- [x] `dialog` should be styled using a chakra `toast` component at the top of the page [tbd]

### Issue 17.
- [x] Tab asset should be added for each category.
- [x] The list items should be filtered and displayed only based on that category.
- [x] Each item will be displayed as an accordion button that includes the `ListItemDetails.jsx` (chakra asset: `AccordionButton`)
  - [x] A `x` icon (chakra asset: `InputRightElement` with `CkClose` chakra icon) will display in the accordion button and perform the `onClick` delete functionality
  - [x] Multiple items should be able to be expanded at once (`allowMultiple`)
- [x] Each item in the accordion button will have a checkbox color based on the urgency category (see wireframe styling)
- [x] Confirm alert that pops up when user deletes item should be styled using the chakra `AlertDialog` component instead of window.confirm

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
| ✓   | :sparkles: New feature     |
| ✓   | :hammer: Refactoring       |


## Updates

### Before
![Screen Shot 2023-05-26 at 9 36 53 PM](https://github.com/the-collab-lab/tcl-55-smart-shopping-list/assets/91443498/fee8b0b2-dfd0-45d3-acb2-505fd72d4b40)


### After
![image](https://github.com/the-collab-lab/tcl-55-smart-shopping-list/assets/99099243/79777c19-1da0-465d-9956-0313eff7267a)


## Testing Steps / QA Criteria

In your terminal, use `git checkout yj-ky-list-item-chakra` to create a new branch and switch over.
Use `git pull origin yj-ky-list-item-chakra` to bring the changes from remote to local.
`npm start` to start a local instance of our app

- Select the `Copy` icon next to the List token name and paste it somewhere to make sure the list is copied correctly onto the clipboard.
- When typing an item into the `Filter Items` input field, the list on whatever tab you're on should update along with all other tabs if you click through each category.
  - When you click on the `X` after your search, the lists should go back to the original items in each category and the input field should go back to the placeholder.
  - You should be able to hover over each Item in the list and click on it to drop down an accordion
    - The Item details should display below the item and you should be able to click the item again to close it
  - Clicking on the garbage can icon should open an Alert Dialog
  - Selecting `Cancel` will bring you back, selecting `Delete` will delete the item in firebase, remove it from the list, and a `Toast` should pop up if successful
- When the list is empty, the tabs and accordion should not display
